core/src/main/java/c/org/rajawali3d/surface/gles/BaseGlesSurfaceRenderer.java

```diff
@@ -1,30 +1,35 @@
 package c.org.rajawali3d.surface.gles;
 
+import c.org.rajawali3d.control.CoreControl;
+import c.org.rajawali3d.control.RenderContextType;
+import c.org.rajawali3d.control.RenderControlClient;
+import c.org.rajawali3d.control.RenderSurfaceView;
+import c.org.rajawali3d.control.gles.GlesRenderControl;
+import c.org.rajawali3d.gl.Capabilities;
+import c.org.rajawali3d.surface.SurfaceRenderer;
+
 import android.content.Context;
 import android.opengl.GLES20;
-import c.org.rajawali3d.core.ARenderControl;
-import c.org.rajawali3d.core.RenderContextType;
-import c.org.rajawali3d.core.RenderControlClient;
-import c.org.rajawali3d.core.RenderSurfaceView;
-import c.org.rajawali3d.gl.Capabilities;
+
 import org.rajawali3d.util.RajLog;
 
 import java.util.Locale;
 
 /**
- * Shared implementation for GL ES extensions of {@link ARenderControl}
+ * Shared implementation for GLES {@link SurfaceRenderer} extensions of {@link CoreControl}
  *
  * @author Randy Picolet
  */
-abstract class GLESRenderer extends ARenderControl {
 
-    GLESRenderer(Context context, RenderSurfaceView renderSurfaceView,
-                 RenderControlClient renderControlClient, double initialFrameRate) {
+abstract class BaseGlesSurfaceRenderer extends CoreControl implements GlesRenderControl {
+
+    BaseGlesSurfaceRenderer(Context context, RenderSurfaceView renderSurfaceView,
+                            RenderControlClient renderControlClient, double initialFrameRate) {
         super(context, renderSurfaceView, renderControlClient, initialFrameRate);
     }
 
     /**
-     * Callback to notify that the EGL context has been acquired and a valid GL thread with EGL surface now exists.
+     *
      */
     protected void onRenderContextAcquired() {
```

Comment:
I know this is from old code, but the call to `Capabilities.getInstance()` here is redundant as we have established it in the `configure()` method of the surfaces.

Reply:
I'm missing where else the Capabilities singleton gets created in the startup sequence...  OTOH, I'm not sure what is gained by instantiating it if it is never used, so maybe this call to getInstance() should just be deleted?

Reply:
My point was to remove it. It used to be used for populating the capabilities once a render context existed but I have changed it to no longer require that.

Reply:
Gone